pep/app/services/pep-session.ts

```diff
@@ -0,0 +1,27 @@
+import session from 'ember-simple-auth/services/session';
+import { oneWay } from '@ember/object/computed';
+import { PepSecureAuthenticatedData } from 'pep/api';
+
+export interface AuthenticatedData {
+    authenticated: PepSecureAuthenticatedData;
+}
+/**
+ * We extend the ember-simple-auth session to strongly type the data object thats stored in it since
+ * we need a custom format
+ *
+ * @export
+ * @class Session
+ * @extends 'ember-simple-auth/services/session
+ */
+export default class Session extends session.extend({
+    data: (oneWay('session.content') as unknown) as AuthenticatedData
+}) {
+    // normal class body definition here
+}
+
+// DO NOT DELETE: this is how TypeScript knows how to look up your services.
+declare module '@ember/service' {
+    interface Registry {
+        'pep-session': Session;
+    }
+}
```

Comment:
we should rename this class to `PepSessionService` and import it by the same name elsewhere, just to be consistent with all the other service's "Session" suffix and matching the filename to the class name.

I can do this in my branch tho once I merge this into mine, if you want. as I already will have to fix a few conflicts I see.